src/PhpSpreadsheet/Cell.php

```diff
@@ -410,7 +410,53 @@ public function setDataValidation(Cell\DataValidation $pDataValidation = null)
     }
 
     /**
-     *    Does this cell contain a Hyperlink?
+     * Does this cell contain valid value?
+     *
+     * @return bool
+     */
+    public function hasValidValue()
+    {
+        if (!$this->hasDataValidation()) {
+            return true;
+        }
+
+        $cellValue = $this->getValue();
+        $cellValidation = $this->getDataValidation();
+
+        if (!$cellValidation->getAllowBlank() && ($cellValue === null || $cellValue == '')) {
+            return false;
+        }
+
+        // TODO: write check on all cases
+        switch ($cellValidation->getType()) {
+            case Cell\DataValidation::TYPE_LIST:
+                $formula1 = $cellValidation->getFormula1();
```

Comment:
We should extract the validation specific for a type into their own method. That would prevent ` hasValidValue()` to become too huge in the future when we add validation for other types.

Reply:
done.